### PR TITLE
Updated baas-test-server to use the arm64 docker image

### PR DESCRIPTION
**Pull request by kraenhansen**
## What, How & Why?

We still need to update the instructions for how to get the AWS credentials, but with this PR the BaaS test server will be pulled from AWS ECR and will run in an Arm64 compatible docker container 🎉

### CODE REVIEW

File: integration-tests/baas-test-server/index.ts [hunk 1]
```diff
@@ -17,40 +17,24 @@
...
-const staticPath = path.resolve(baasPath, "static");
-const staticAppPath = path.resolve(staticPath, "app");
-const dylibPath = path.resolve(baasPath, "etc/dylib");
-const transpilerPath = path.resolve(baasPath, "etc/transpiler");
-const transpilerBinPath = path.resolve(transpilerPath, "bin");
-const baasTmpPath = path.resolve(baasPath, "tmp");
-const dylibFilename = "libstitch_support.dylib";
-const dylibUsrLocalLibPath = "/usr/local/lib/" + dylibFilename;
-const assistedAggPath = path.resolve(baasPath, "assisted_agg");
+const { AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY } = process.env;
```
Comment (kneth) [inline integration-tests/baas-test-server/index.ts:34]: Does it mean that you don't need to store them in `.env`?

**Review by kneth — APPROVED** [review]
LGTM

File: integration-tests/baas-test-server/index.ts (hunk 1 above)
Comment (kraenhansen) [reply integration-tests/baas-test-server/index.ts:34]: Yep - or provide them to the process on the command-line, but the `dotenv.config()` above helps load them from a file.

File: integration-tests/baas-test-server/index.ts [hunk 2]
```diff
@@ -90,223 +69,144 @@ function ensureDocker() {
...
+  taskId: string;
+  execution: number;
+  timestamp: string;
+  branch: string;
+  revision: string;
+  */
+};
+
+function assertImage(value: unknown): asserts value is Image {
+  assert(typeof value === "object" && value !== null);
```
Comment (elle-j) [inline integration-tests/baas-test-server/index.ts:97]:
```suggestion
  assert(typeof value === "object" && value !== null && !Array.isArray(value));
```

File: integration-tests/baas-test-server/index.ts [hunk 3]
```diff
@@ -90,223 +69,144 @@ function ensureDocker() {
...
+  assert(typeof object.imageTag === "string");
+}
+
+type ImagesResponse = {
+  allBranches: string[];
+  images: Record<string, undefined | Image[]>;
+};
+
+function assertImagesResponse(value: unknown): asserts value is ImagesResponse {
+  assert(typeof value === "object" && value !== null);
```
Comment (elle-j) [inline integration-tests/baas-test-server/index.ts:109]:
```suggestion
  assert(typeof value === "object" && value !== null && !Array.isArray(value));
```

File: integration-tests/baas-test-server/index.ts [hunk 4]
```diff
@@ -90,223 +69,144 @@ function ensureDocker() {
...
+  const object = value as Record<string, unknown>;
+  assert(Array.isArray(object.allBranches));
+  const { images } = object;
+  assert(typeof images === "object" && images !== null);
```
Comment (elle-j) [inline integration-tests/baas-test-server/index.ts:113]:
```suggestion
  assert(typeof images === "object" && images !== null && !Array.isArray(images));
```

**Review by elle-j — APPROVED** [review]
-100 lines 🎉  Nice!

File: integration-tests/baas-test-server/index.ts (hunk 2 above)
Comment (kraenhansen) [reply integration-tests/baas-test-server/index.ts:97]: I think it's fine. Without the explicit check for array, the property checks below will return undefined and fail their assertions.

**Pull request merged**